sklearn/decomposition/_sparse_pca.py

```diff
@@ -29,6 +29,66 @@ def _check_normalize_components(normalize_components, estimator_name):
             )
 
 
+def _get_explained_variance(X, components):
+    '''Get the explained variance.
+
+    Get the explained variance from the principal components of the
+    data. This follows the method outlined in [1] section 3.4 (Adjusted Total
+    Variance). For an alternate approach (not implemented here), see [2].
+
+    Parameters
+    ----------
+    X : ndarray, shape (n_samples, n_features)
+        The feature vector. n_samples and n_features are the number of
+        samples and features, respectively.
+
+    components : array, shape (n_components, n_features)
+        The (un-normalized) principle components. [1]
+
+    Notes
+    -----
+    The variance ratio may not be computed. The main reason is that we
+    do not know what the total variance is since we did not compute all
```

Comment:
Maybe rephrase to.

> The variance ratio cannot be computed. Indeed, we
> do not know the total variance since we did not compute all
> the components.